src/main/java/org/apache/commons/lang3/time/DateUtils.java

```diff
@@ -1625,6 +1628,34 @@ public static Calendar toCalendar(final Date date, final TimeZone tz) {
         return c;
     }
 
+    /**
+     * Converts a {@link Date} into a {@link LocalDateTime}, using the default time zone.
+     * @param date the date to convert to a LocalDateTime
+     * @return the created LocalDateTime
+     * @throws NullPointerException if {@code date} is null
+     * @since 3.18
+     */
+    public static LocalDateTime toLocalDateTime(final Date date) {
+        return toLocalDateTime(date, TimeZone.getDefault());
+    }
+
+    /**
+     * Converts a {@link Date} into a {@link LocalDateTime}
+     * @param date the date to convert to a LocalDateTime
+     * @param tz the time zone of the {@code date}
+     * @return the created LocalDateTime
+     * @throws NullPointerException if {@code date} is null
+     * @since 3.18
+     */
+    public static LocalDateTime toLocalDateTime(final Date date, final TimeZone tz) {
+        final LocalDateTime localDateTime = LocalDateTime.ofInstant(Instant.ofEpochMilli(Objects.requireNonNull(date, "date").getTime()),
+                Objects.requireNonNull(tz, "tz").toZoneId());
+        if (date instanceof java.sql.Timestamp) {
```

Comment:
This dependency is not OK because it brings in a module we did not depend on in the past, `java.sql`, and we've had concerns in the past that we already depend on the `java.desktop` module, but we can't change that without breaking apps in surprising ways. We should do our best for now to avoid bringing in modules outside of `java.base`.